fdb-relational-core/src/main/java/com/apple/foundationdb/relational/recordlayer/metadata/RecordLayerSchemaTemplate.java

```diff
@@ -476,6 +476,15 @@ public Builder replaceInvokedRoutine(@Nonnull final RecordLayerInvokedRoutine in
             return this;
         }
 
+        public Builder removeInvokedRoutine(@Nonnull final String invokedRoutineName) {
+            if (!invokedRoutines.containsKey(invokedRoutineName)) {
+                Assert.thatUnchecked(invokedRoutines.get(invokedRoutineName).isTemporary(), ErrorCode.UNDEFINED_FUNCTION,
+                        "attempt to non-existent temporary invoked routine!");
+            }
+            invokedRoutines.remove(invokedRoutineName);
+            return this;
+        }
```

Comment:
I think this method should _just_ remove the invoked routine whose name matches the given argument. Checking whether the routine is actually temporary or not is the responsibility of the customer, which is your case is the  `DropTemporaryFunctionConstantAction`.